services/analytics_service.py

```
@@ -1,399 +1,349 @@
+        # Get raw data and process it in Python for more control
+        applications = db.session.query(
+            JobApplication.last_update,
+            JobApplication.id
+        ).filter(
+            JobApplication.last_update >= twelve_weeks_ago
+        ).all()
+        
+        # Group by week in Python
+        weekly_counts = defaultdict(int)
+        
+        for app_date, app_id in applications:
```

Ruff (0.12.2)

Loop control variable `app_id` not used within loop body

Rename unused `app_id` to `_app_id`

(B007)